laalaa/apps/advisers/importer.py

```diff
@@ -84,12 +98,26 @@ def __init__(self, path):
         self.category_civil_sheet = workbook.sheet_by_name('CAT OF LAW CIVIL')
         self.outreach_sheet = workbook.sheet_by_name('OUTREACH SERVICE')
 
-    def sheet_to_dict(self, worksheet):
+    def interrupt(self):
+        self._interrupt.set()
+
+    def check_interrupt(self):
+        if self._interrupt.is_set():
+            raise KeyboardInterrupt
+
+    @classmethod
+    def sheet_to_dict(cls, worksheet):
         """
         Parse worksheet into list of dicts
         """
+        class StrippedDict(dict):
+            def __getitem__(self, key):
+                item = super(StrippedDict, self).__getitem__(key)
+                if isinstance(item, basestring):
+                    return item.strip()
+                return item
```

Comment:
Perhaps this would be better to override `__setitem__` - so that you strip the value just once when you add it, instead of every time you access it.